src/xg.cpp

```diff
@@ -2221,7 +2379,23 @@ void XG::expand_context_by_steps(Graph& g, size_t steps, bool add_paths,
                     edges[sides] = ep;
                 }
             }
-        }       
+        }
+        for (auto& edge : edges_to(n)) {
+            if(last_step_nodes.count(edge.from())) {
+                // This edge connects two nodes that were added on the last
+                // step, and so wouldn't have been found by the main loop.
+                
+                // Determine if it's been seen already (somehow).
+                // TODO: it probably shouldn't have been, unless it's a self loop or something.
+                auto sides = make_pair(make_side(edge.from(), edge.from_start()),
```

Comment:
First, the first invocation of `make_side` looks wrong to me. I think it should be `!edge.from_start()`.

Second, this isn't a canonical representation. The edge could equally validly be represented as 
```
make_pair(edge.to(), edge.to_end())
          edge.from(), !edge.from_start()))
```
That means you can't count on the `edges` set to identify duplicates, because you might have represented the same edge in two different ways (so the `==` operator would return false). You need to break the symmetry and add a method to always choose the same canonical way of representing an edge. Typically, this involves choosing the representation that's lowest among some sort order. For instance, so that the lower node ID is always the first one in the pair (but be careful about reversing and non-reversing self-edges).